pkg/credentials/generator.go

```diff
@@ -0,0 +1,141 @@
+package credentials
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/deislabs/duffle/pkg/bundle"
+	"github.com/deislabs/duffle/pkg/credentials"
+
+	survey "gopkg.in/AlecAivazis/survey.v1"
+)
+
+type GenerateOptions struct {
+
+	// Name of the credential.
+	Name string
+
+	//Credentials from the bundle
+	Credentials map[string]bundle.Location
+
+	//Should we survey?
+	Silent bool
+}
+
+type credentialAnswers struct {
+	Source string `survey:"source"`
+	Value  string `survey:"value"`
+}
+
+const (
+	questionValue   = "specific value"
+	questionEnvVar  = "environment variable"
+	questionPath    = "file path"
+	questionCommand = "shell command"
+)
+
+type credentialGenerator func(name string) (credentials.CredentialStrategy, error)
+
+// GenerateCredentials will generate a credential set based on the given options
+func GenerateCredentials(opts GenerateOptions) (*credentials.CredentialSet, error) {
+	if opts.Name == "" {
+		return nil, errors.New("credentialset name is required")
+	}
+	generator := genCredentialSurvey
+	if opts.Silent {
+		generator = genEmptyCredentials
+	}
+	credSet, err := genCredentialSet(opts.Name, opts.Credentials, generator)
+	if err != nil {
+		return nil, err
+	}
+	return &credSet, nil
+}
+
+func genCredentialSet(name string, creds map[string]bundle.Location, fn credentialGenerator) (credentials.CredentialSet, error) {
+	cs := credentials.CredentialSet{
+		Name: name,
+	}
+	cs.Credentials = []credentials.CredentialStrategy{}
+
+	if strings.ContainsAny(name, "./\\") {
+		return cs, fmt.Errorf("credentialset name '%s' cannot contain the following characters: './\\'", name)
+	}
+
+	var credentialNames []string
+	for name := range creds {
+		credentialNames = append(credentialNames, name)
+	}
+
+	sort.Strings(credentialNames)
+
+	for _, name := range credentialNames {
+		c, err := fn(name)
+		if err != nil {
+			return cs, err
+		}
+		cs.Credentials = append(cs.Credentials, c)
+	}
+
+	return cs, nil
+}
+
+func genEmptyCredentials(name string) (credentials.CredentialStrategy, error) {
+	return credentials.CredentialStrategy{
+		Name:   name,
+		Source: credentials.Source{Value: "EMPTY"},
```

Comment:
Can we use TODO instead of EMPTY? Just to drive home that it's not filled in?

Reply:
sure!